### PR TITLE
fix: undefined validation object fix

### PR DESCRIPTION
**Pull request by Nandana-NNR**

### CODE REVIEW

**Review by ibrahimshuail — APPROVED** [review]

File: cypress/support/step_definitions/validations.js [hunk 1]
```diff
@@ -177,30 +177,37 @@ Then(/'(.+)' will (be|stay) in '(.+)' state/, (app, condition, state) => {
...
+          );
+        }
       }
     }
-  } else {
+  } catch (error) {
+    throw new Error(
+      `Following error occurred during custom validation: ${JSON.stringify(error.message)}`
+    );
+  }
```
Comment (neeradanelxsi) [inline cypress/support/step_definitions/validations.js:209]: i think we should only print error here, right? suppose if you get an error and there is no message field?

Comment (neeradanelxsi) [inline cypress/support/step_definitions/validations.js:209]: also, is there an else condition for line 185 ? What if there is no validationObject ?

File: cypress/support/step_definitions/validations.js [hunk 2]
```diff
@@ -177,30 +177,37 @@ Then(/'(.+)' will (be|stay) in '(.+)' state/, (app, condition, state) => {
...
+  if (!validationObject) {
```
Comment (neeradanelxsi) [inline cypress/support/step_definitions/validations.js:210]: not really understanding the purpose of moving else condition outside to line 210. Cant we do 
```

try {
      if (validationObject) {
        changes from line 186 to 204
      }else{
         changes from lines 211 to 216
       }
}catch(error) {
     Here, give a generic error message 
     throw new Error(
      `Error occurred during validation: ${JSON.stringify(error)}`
    );
}
```

**Review by jbigel — DISMISSED** [review]

File: cypress/support/step_definitions/validations.js (hunk 2 above)
Comment (Nandana-NNR) [reply cypress/support/step_definitions/validations.js:210]: Addressed

File: cypress/support/step_definitions/validations.js [hunk 3]
```diff
@@ -177,36 +177,40 @@ Then(/'(.+)' will (be|stay) in '(.+)' state/, (app, condition, state) => {
...
-    cy.validateLifecycleState(appObject.getAppObjectState().state, appId);
-    cy.validateLifecycleHistoryAndEvents(
-      appObject.getAppObjectState().state,
-      appId,
-      isEventsExpected
-    );
+  } catch (error) {
+    throw new Error(`Following error occurred during custom validation: ${JSON.stringify(error)}`);
   }
 });
```
Comment (neeradanelxsi) [inline cypress/support/step_definitions/validations.js:215]: throw new Error(`Following error occurred during custom validation: ${JSON.stringify(error)}`); to be changed to 
throw new Error(`Error occurred during validation: ${JSON.stringify(error)}`);

**Review by neeradanelxsi — APPROVED** [review]
Minor comment on language for log

**Review by neeradanelxsi — APPROVED** [review]

**Pull request merged**